libs/interpret/src/lib/MLIDashboard/Controls/GlobalExplanationTab/GlobalExplanationTab.tsx

```diff
@@ -181,6 +181,10 @@ export class GlobalExplanationTab extends React.PureComponent<
       this.context.jointDataset.datasetMetaData?.featureMetaData
         ?.dropped_features
     );
+    const selectedCohort =
+      this.state.selectedCohortIndex >= this.props.cohorts.length
+        ? this.props.cohorts[0]
+        : this.props.cohorts[this.state.selectedCohortIndex];
```

Comment:
So this captures the case of going beyond the bounds, but if that's possible that indicates to me that we're not updating properly. So let's say there are cohorts `[0,1,2]` and we delete 2 then that's your case here, we fix that by going to 0. Great! But if we delete 1 then that would put us to.... 2? That seems unintuitive, doesn't it? 
There's a very decent chance that I'm misunderstanding what's going on here, though 😄

Reply:
Agree that this happens because we are not updating properly. This also happened in other component for undefined cohorts. I am thinking if it is more user-friendly by disabling deletion of current selected cohort? So user won't be confused by we auto select another cohort when they delete the current one.

Reply:
I think we are already doing this. We won't let users delete current cohort.

Reply:
good points! I updated the files so that we calculate the selected cohort when a cohort is deleted. This would take care of the case you mentioned @romanlutz so the selected cohort persists. In the case that the user deleted the cohort that they are currently viewing, right now I have the selected cohort to default to the All data default cohort since cohort is required to be defined in FeatureImportanceDependence